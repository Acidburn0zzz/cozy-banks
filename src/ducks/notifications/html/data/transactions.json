[
  {
    "account": "compteisa1",
    "amount": -60,
    "categoryId": "400610",
    "currency": "€",
    "date": "2017-07-06T00:00:00Z",
    "demo": true,
    "label": "CABINET OSTEOKINESITHERAPIE",
    "reimbursements": [
      {
        "amount": 60,
        "operationId": "avrilop.sante.isa2",
        "vendor": "Harmonie"
      }
    ]
  },
  {
    "account": "compteisa1",
    "amount": -60,
    "categoryId": "400610",
    "currency": "€",
    "date": "2017-07-06T00:00:00Z",
    "demo": true,
    "label": "CABINET OSTEOKINESITHERAPIE",
    "reimbursements": [
      {
        "amount": 60,
        "operationId": "avrilop.sante.isa2",
        "vendor": "Harmonie"
      }
    ]
  },
  {
    "account": "comptecla1",
    "amount": 28.7,
    "bill": "io.cozy.bills:avrilrembourcla3",
    "categoryId": "400610",
    "currency": "€",
    "date": "2017-07-11T00:00:00Z",
    "demo": true,
    "label": "CPAM PARIS"
  },
  {
    "account": "compteisa3",
    "amount": 400,
    "categoryId": "600110",
    "currency": "€",
    "date": "2017-07-03T00:00:00Z",
    "demo": true,
    "label": "TRANSFERT LIVRET A"
  },
  {
    "account": "comptegene1",
    "amount": -42.3,
    "categoryId": "400110",
    "currency": "€",
    "date": "2017-07-04T00:00:00Z",
    "demo": true,
    "label": "AUCHAN"
  },
  {
    "account": "compteisa1",
    "amount": 7.5,
    "bill": "io.cozy.bills:rembourcomplisa1",
    "categoryId": "400610",
    "currency": "€",
    "date": "2017-08-25T00:00:00Z",
    "demo": true,
    "label": "VIREMENT COMPL SANTE"
  }
]
