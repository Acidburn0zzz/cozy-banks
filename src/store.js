import thunkMiddleware from 'redux-thunk'
import { createStore, applyMiddleware, compose } from 'redux'
import createLogger from 'redux-logger'
import appReducers from 'reducers'

import {
  shouldEnableTracking,
  getTracker,
  createTrackerMiddleware } from 'cozy-ui/react/helpers/tracker'

const composeEnhancers = (__DEVELOPMENT__ && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose

const loggerMiddleware = createLogger()

const middlewares = [
  thunkMiddleware,
  loggerMiddleware
]

if (shouldEnableTracking() && getTracker()) {
  middlewares.push(createTrackerMiddleware())
}

export default createStore(
  appReducers,
  composeEnhancers(applyMiddleware.apply(null, middlewares))
)
