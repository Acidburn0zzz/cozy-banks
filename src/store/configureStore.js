/* global PouchDB, pouchdbFind */
/* global __DEVELOPMENT__ __TARGET__ */
import { compose, createStore, applyMiddleware } from 'redux'
import thunkMiddleware from 'redux-thunk'
import { createLogger } from 'redux-logger'
import { shouldEnableTracking, getTracker, createTrackerMiddleware } from 'cozy-ui/react/helpers/tracker'

import { ACCOUNT_DOCTYPE, GROUP_DOCTYPE, TRANSACTION_DOCTYPE } from 'doctypes'
import appReducers from 'reducers'
import { cozyMiddleware, CozyClient } from 'redux-cozy-client'

const getCozyClient = function () {
  // Bind `PouchDB` to window for `cozy-client-js` to find it
  // PouchDB is provided here by `webpack` through `ProvidedPlugin`
  window.PouchDB = PouchDB
  window.pouchdbFind = pouchdbFind

  const cozyOptions = {
    offline: { doctypes: [ACCOUNT_DOCTYPE, GROUP_DOCTYPE, TRANSACTION_DOCTYPE] }
  }

  if (__TARGET__ === 'browser') {
    const root = document.querySelector('[role=application]')
    const data = root.dataset
    cozyOptions.cozyURL = `${window.location.protocol}//${data.cozyDomain}`
    cozyOptions.token = data.cozyToken
  } else {
    cozyOptions.cozyURL = persistedUrl
  }

  return new CozyClient(cozyOptions)
}

const configureStore = persistedState => {
  // Enable Redux dev tools
  const composeEnhancers = (__DEVELOPMENT__ && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose

  // reducers
  const reducers = [appReducers, persistedState]

  // middlewares
  const middlewares = [
    thunkMiddleware,
    cozyMiddleware(getCozyClient(persistedState && persistedState.mobile.url ? persistedState.mobile.url : null))
  ]
  if (shouldEnableTracking() && getTracker()) {
    middlewares.push(createTrackerMiddleware())
  }
  if (__DEVELOPMENT__) {
    // must be the last middleware in chain https://git.io/vHQpt
    const loggerMiddleware = createLogger()
    middlewares.push(loggerMiddleware)
  }

  const store = createStore(
    ...reducers,
    composeEnhancers(applyMiddleware.apply(null, middlewares))
  )

  return store
}

export default configureStore
