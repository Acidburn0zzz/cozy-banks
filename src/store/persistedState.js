import localforage from 'localforage'

export const persistState = store => {
  store.subscribe(() => saveState({
    filters: {
      accountOrGroupType: store.getState().filters.accountOrGroupType,
      accountOrGroupId: store.getState().filters.accountOrGroupId
    },
    mobile: store.getState().mobile
  }))
}

export const loadState = async () => {
  try {
    const persistedState = await localforage.getItem('state')
    if (persistedState === null) {
      return undefined
    }
    return persistedState
  } catch (err) {
    console.warn(err)
    return undefined
  }
}

export const saveState = async (state) => {
  try {
    localforage.setItem('state', state)
  } catch (err) {
    console.warn(err)
    // Errors handling
  }
}
