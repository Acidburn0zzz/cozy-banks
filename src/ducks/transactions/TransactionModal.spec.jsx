/* global mount */

import React from 'react'
import { DumbTransactionModal as TransactionModal } from './TransactionModal'
import data from '../../../test/fixtures'
import AppLike from 'test/AppLike'
import store from 'test/store'
import pretty from 'pretty'

jest.mock('preact-portal', () => ({ children }) => children)

const allTransactions = data['io.cozy.bank.operations']
const allAccounts = data['io.cozy.bank.accounts']

describe('transaction modal', () => {
  it('should render correctly', () => {
    // need to fix number of the account otherwise its randomly
    // set by the fixture
    const transaction = { ...allTransactions[0], _id: '2' }
    const account = { ...allAccounts[0], number: 1 }
    const root = mount(
      <AppLike store={store}>
        <TransactionModal
          transactionId={transaction._id}
          transaction={transaction}
          account={account}
          showCategoryChoice={() => {}}
          requestClose={() => {}}
          urls={{}}
          brands={[]}
        />
      </AppLike>
    )
    expect(pretty(root.html())).toMatchSnapshot()
  })
})
