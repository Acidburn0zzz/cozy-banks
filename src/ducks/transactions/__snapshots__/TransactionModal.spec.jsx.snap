// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transaction modal should render correctly 1`] = `
"<div class=\\"c-modal-container\\">
  <div class=\\"c-overlay\\">
    <div class=\\"c-modal-wrapper\\">
      <div class=\\"c-modal c-modal--small TransactionModal c-modal--closable\\">
        <button type=\\"submit\\" role=\\"button\\" class=\\"c-modal-close TransactionModalCross c-btn c-btn--close c-btn--narrow\\"><span><svg class=\\"icon\\" width=\\"24\\" height=\\"24\\" style=\\"fill: #95999D;\\"><use xlink:href=\\"test-file-stub\\"></use></svg></span></button>
        <div class=\\"c-modal-header TransactionModalHeader\\">
          <h2 class=\\"TransactionModalHeading\\">
            <div class=\\"\\"><span class=\\"Figure-total\\"><!-- react-text: 13 -->-1,231.00<!-- /react-text --></span><span class=\\"Figure-currency Figure__currency--withSpacing\\">â‚¬</span></div>
          </h2>
        </div>
        <div class=\\"c-modal-content TransactionModalContent\\">
          <div>
            <div class=\\"media TransactionModalRow TransactionModalRow-multiline media--top\\">
              <div class=\\"img TransactionModalRowIcon\\">
                <!-- react-empty: 19 -->
              </div>
              <div class=\\"bd TransactionModalRowContent\\">
                <div class=\\"TransactionLabel\\">Remboursement Pret Lcl</div>
                <div class=\\"TransactionInfos\\">
                  <div class=\\"TransactionInfo\\"><span class=\\"TransactionInfoLabel\\"><!-- react-text: 25 -->Account<!-- /react-text --><!-- react-text: 26 --> :<!-- /react-text --></span>
                    <!-- react-text: 27 -->PEE Isabelle
                    <!-- /react-text -->
                  </div>
                  <div class=\\"TransactionInfo\\"><span class=\\"TransactionInfoLabel\\"><!-- react-text: 30 -->Bank<!-- /react-text --><!-- react-text: 31 --> :<!-- /react-text --></span>
                    <!-- react-text: 32 -->HSBC
                    <!-- /react-text -->
                  </div>
                </div>
              </div>
            </div>
            <hr class=\\"TransactionModalSeparator\\">
            <div class=\\"media TransactionModalRow\\">
              <div class=\\"img TransactionModalRowIcon\\">
                <!-- react-empty: 36 -->
              </div>
              <div class=\\"bd TransactionModalRowContent\\"><span style=\\"text-transform: capitalize;\\">Friday 25 August</span></div>
            </div>
            <hr class=\\"TransactionModalSeparator\\">
            <div class=\\"media TransactionModalRow TransactionModalRow-clickable\\">
              <div class=\\"img TransactionModalRowIcon\\">
                <!-- react-empty: 42 -->
              </div>
              <div class=\\"bd TransactionModalRowContent\\">Real estate loan</div>
              <div class=\\"img\\">
                <!-- react-empty: 45 -->
              </div>
            </div>
            <hr class=\\"TransactionModalSeparator\\">
            <div class=\\"media TransactionModalRow TransactionModalRow-disabled\\">
              <div class=\\"img TransactionModalRowIcon\\">
                <!-- react-empty: 49 -->
              </div>
              <div class=\\"bd TransactionModalRowContent\\">Comment - available soon</div>
            </div>
            <hr class=\\"TransactionModalSeparator\\">
            <!-- react-empty: 52 -->
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
